plugins/inputs/icinga2/icinga2_test.go

```diff
@@ -10,36 +15,55 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
-func TestGatherServicesStatus(t *testing.T) {
-	s := `{
-    "results": [
-        {
-            "attrs": {
-                "check_command": "check-bgp-juniper-netconf",
-                "display_name": "eq-par.dc2.fr",
-                "host_name": "someserverfqdn.net",
-                "name": "ef017af8-c684-4f3f-bb20-0dfe9fcd3dbe",
-                "state": 0
-            },
-            "joins": {},
-            "meta": {},
-            "name": "eq-par.dc2.fr!ef017af8-c684-4f3f-bb20-0dfe9fcd3dbe",
-            "type": "Service"
-        }
-    ]
+const icinga2ServiceResponse = `{
+	"results": [
+		{
+			"attrs": {
+				"check_command": "check-bgp-juniper-netconf",
+				"display_name": "eq-par.dc2.fr",
+				"host_name": "someserverfqdn.net",
+				"name": "ef017af8-c684-4f3f-bb20-0dfe9fcd3dbe",
+				"state": 0
+			},
+			"joins": {},
+			"meta": {},
+			"name": "eq-par.dc2.fr!ef017af8-c684-4f3f-bb20-0dfe9fcd3dbe",
+			"type": "Service"
+		}
+	]
 }
 `
 
-	checks := Result{}
-	require.NoError(t, json.Unmarshal([]byte(s), &checks))
+const icinga2HostResponse = `{
+	"results": [
+		{
+			"attrs": {
+				"address": "192.168.1.1",
+				"check_command": "ping",
+				"display_name": "apache",
+				"name": "webserver",
+				"state": 2.0
+			},
+			"joins": {},
+			"meta": {},
+			"name": "webserver",
+			"type": "Host"
+		}
+	]
+}
+`
+
+func TestGatherServicesStatusOld(t *testing.T) {
```

Comment:
This method is not adding any more value versus `TestGatherServicesStatus` or you should use `icinga2.Gather` to test backward compatibility.

Reply:
I've tried not to touch the original tests, but they were not testing the `Gather` method at all. Maybe we can move the common code for the "old" and "new" configuration to a method. Or is the code repetition ok in tests?

Reply:
The problem is now you have 2 tests that are actually doing the same, so there is no benefit of having the second test..

Reply:
Those tests were redundant. Now the config checked in `Init()` method, so testing that the old config is translated into new format should be enough.